lib/SILOptimizer/Utils/SILInliner.cpp

```diff
@@ -128,6 +128,31 @@ class BeginApplySite {
       collectAbortApply(abortApply);
       endBorrowInsertPts.push_back(&*std::next(abortApply->getIterator()));
     }
+
+    // mark_dependence/mark_dependence_addr may have begin_apply's token result
+    // as their operand. This is needed for diagnosing lifetime dependencies for
+    // ~Escapable yields. Delete them while inlining the begin_apply since the
+    // dependence is also represented by lifetime dependence sources.
```

Comment:
I think this comment is a little off...

The mark_dependence on the coroutine token represents a dependency on the coroutine context. That is different from a dependency on one of the coroutine parameters. For example, the yielded value may depend on a value created inside the coroutine.

I guess we can assume that inlining the coroutine will expose any dependencies within the coroutine implementation, so we can delete the dependency on the context after inlining, but I haven't investigated that at all. It would be worth adding a unit test for that case that checks the SIL after inlining.